build.gradle

```
@@ -5,11 +5,11 @@ buildscript {
       'minSdk': 14,
       'compileSdk': 30,
 
-      'kotlin': '1.5.21',
+      'kotlin': '2.1.20',
       'autoService': '1.1.1',
 
       // Update WrongTimberUsageDetectorTest#innerStringFormatWithStaticImport when >= 7.1.0-alpha07
-      'androidPlugin': '7.0.0',
+      'androidPlugin': '8.10.0',
       'androidTools': '30.0.0',
```

This version should always be the previous one plus 23
```suggestion
      'androidTools': '31.10.0',
```

---

Unfortunately, making this change is going to be non-trivial. It appears that the lint rules have changed and updating the android tools causes the lint test to fail. I'm investigating how to minimize the LOC of the change, but I think it's going to be reasonably large.

I would really prefer to not change tests and these dependency numbers at the same time. My preference (if I was working with HG stacks and Phabricator) would be to have this be one diff, and the next diff on the stack would be the test fixes and the tools number bump. We could then review and have CI test each revision individually, and land the stack as a single unit. If this is possible in Git I'm unaware of how to do it.

For the time being, I'll be making the tools update and the test changes in a follow up PR, and will be awaiting your feedback before fixing this one.